modules/core/src/main/java/org/apache/ignite/internal/cdc/WalRecordsConsumer.java

```diff
@@ -89,70 +104,29 @@ public WalRecordsConsumer(CdcConsumer consumer, IgniteLogger log) {
     }
 
     /**
-     * Handles record from the WAL.
-     * If this method return {@code true} then current offset in WAL will be stored and WAL iteration will be
-     * started from it on CDC application fail/restart.
+     * Handles data entries.
+     * If this method return {@code true} then current offset in WAL and {@link DataEntry} index inside
+     * {@link DataRecord} will be stored and WAL iteration will be started from it on CDC application fail/restart.
      *
-     * @param recs WAL records iterator.
+     * @param entries Data entries iterator.
      * @return {@code True} if current offset in WAL should be commited.
      */
-    public boolean onRecords(Iterator<DataRecord> recs) {
-        Iterator<CdcEvent> evts = new Iterator<CdcEvent>() {
-            /** */
-            private Iterator<CdcEvent> entries;
-
+    public boolean onRecords(DataEntryIterator entries) {
```

Comment:
Looks like we can keep `Iterator<DataRecord>` parameter type. Let's do it?

Reply:
Done.